### PR TITLE
Frontend testing

### PR DESCRIPTION
**Pull request by azhara-a**
Write tests for Sign Up and Login pages with Jest Fetch Mock and React Testing Library

### CODE REVIEW

File: frontend/src/components/account/sign-up/SignUp.test.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import SignUp from "./SignUp";
+import { getDietaryRequirements } from "../../../utils/DietaryRequirements";
+
```
Comment (mugdhalakhani) [inline frontend/src/components/account/sign-up/SignUp.test.js:19]: nit: can you add a comment describing what the tests in this file are testing?

File: frontend/src/components/account/sign-up/SignUp.test.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+});
+
+test("renders 'change account details' page", () => {
+  const location = { state: { name: "name", diets: [], allergies: [] } };
+  const { getByText } = render(<SignUp location={location} />);
+  const title = getByText("Change account details");
+  expect(title).toBeInTheDocument();
+});
+
+test("displays user diets if 'change account details' page", () => {
```
Comment (mugdhalakhani) [inline frontend/src/components/account/sign-up/SignUp.test.js:34]: this isn't very clear. Is this shown when the user tries to sign up but already has an account?

File: frontend/src/components/account/sign-up/SignUp.test.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import { getDietaryRequirements } from "../../../utils/DietaryRequirements";
+
+test("renders 'sign up' page", () => {
+  const location = { state: undefined };
+  const { getByText } = render(<SignUp location={location} />);
+  const title = getByText("Sign Up");
+  expect(title).toBeInTheDocument();
+});
+
+test("renders 'change account details' page", () => {
```
Comment (mugdhalakhani) [inline frontend/src/components/account/sign-up/SignUp.test.js:27]: Hmm, but we're rendering the signup page on line 29.

**Review by mugdhalakhani — COMMENTED** [review]
Thanks Azhara. Some questions in Signup tests, the rest looks good!

File: frontend/src/components/account/sign-up/SignUp.test.js (hunk 3 above)
Comment (azhara-a) [reply frontend/src/components/account/sign-up/SignUp.test.js:27]: I use SignUp.js for both sign up and changing account details. The difference is that during 'sign up' the form is empty, but during 'change account details' the form is pre-filled with user details.

File: frontend/src/components/account/sign-up/SignUp.test.js (hunk 1 above)
Comment (azhara-a) [reply frontend/src/components/account/sign-up/SignUp.test.js:19]: Yup

File: frontend/src/components/account/sign-up/SignUp.test.js (hunk 2 above)
Comment (azhara-a) [reply frontend/src/components/account/sign-up/SignUp.test.js:34]: I am testing whether user data gets pre-filled correctly in the change account details page. I'll reword the test description

**Review by mam249 — APPROVED** [review]

File: frontend/src/components/account/sign-up/SignUp.test.js (hunk 3 above)
Comment (mugdhalakhani) [reply frontend/src/components/account/sign-up/SignUp.test.js:38]: ack, thanks.

**Review by mugdhalakhani — APPROVED** [review]

**Pull request merged**